delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/organizations/OrganizationsImpl.scala

```diff
@@ -0,0 +1,257 @@
+        )
+      }
+      .named("listOrganizations", component)
+
+  override def events(offset: Offset): fs2.Stream[Task, Envelope[OrganizationEvent]] =
+    eventLog.eventsByTag(entityType, offset)
+
+  override def currentEvents(offset: Offset): fs2.Stream[Task, Envelope[OrganizationEvent]] =
+    eventLog.currentEventsByTag(entityType, offset)
+
+}
+
+object OrganizationsImpl {
+
+  type OrganizationsAggregate =
+    Aggregate[String, OrganizationState, OrganizationCommand, OrganizationEvent, OrganizationRejection]
+
+  type OrganizationsCache = KeyValueStore[Label, OrganizationResource]
+
+  /**
+    * The organizations entity type.
+    */
+  final val entityType: String = "organizations"
+
+  /**
+    * The organizations tag name.
+    */
+  final val organizationTag = "organization"
+
+  private val logger: Logger = Logger[OrganizationsImpl]
+
+  /**
+    * Creates a new organization cache.
+    */
+  private def cache(config: OrganizationsConfig)(implicit as: ActorSystem[Nothing]): OrganizationsCache = {
+    implicit val cfg: KeyValueStoreConfig           = config.keyValueStore
+    val clock: (Long, OrganizationResource) => Long = (_, resource) => resource.rev
+    KeyValueStore.distributed("organizations", clock)
+  }
+
+  private def startIndexing(
+      config: OrganizationsConfig,
+      eventLog: EventLog[Envelope[OrganizationEvent]],
+      index: OrganizationsCache,
+      organizations: Organizations
+  )(implicit as: ActorSystem[Nothing], sc: Scheduler) = {
+    val singletonManager = ClusterSingleton(as)
+    singletonManager.init(
+      SingletonActor(
+        Behaviors
+          .supervise(
+            StreamSupervisor.behavior(
+              Task.delay(
+                eventLog
+                  .eventsByTag(organizationTag, Offset.noOffset)
+                  .mapAsync(config.indexing.concurrency)(envelope =>
+                    organizations.fetch(envelope.event.label).flatMap {
+                      case Some(org) => index.put(org.id, org)
+                      case None      => UIO.unit
+                    }
+                  )
+              ),
+              RetryStrategy(
+                config.indexing.retryStrategy,
+                _ => true,
+                RetryStrategy.logError(logger, "organizations indexing")
+              )
+            )
+          )
+          .onFailure[Exception](SupervisorStrategy.restart),
+        "OrganizationsIndex"
+      )
+    )
+  }
+
+  private def aggregate(
+      config: OrganizationsConfig
+  )(implicit as: ActorSystem[Nothing], clock: Clock[UIO]): UIO[OrganizationsAggregate] = {
+    val definition = PersistentEventDefinition(
+      entityType = entityType,
+      initialState = OrganizationState.Initial,
+      next = Organizations.next,
+      evaluate = Organizations.evaluate,
+      tagger = (_: OrganizationEvent) => Set(entityType),
+      snapshotStrategy = SnapshotStrategy.SnapshotCombined(
```

Comment:
we should probably have the snapshot strategy for all resource types as part of the config; happy to have that in a separate PR.

Reply:
Yes, let's have it as another PR